packages/server/test/e2e/8_read_only_fs.ts

```diff
@@ -0,0 +1,55 @@
+import * as fs from 'fs'
+import * as path from 'path'
+
+const e2e = require('../support/helpers/e2e')
+const chmodr = require('chmodr')
+
+describe('e2e project dir access', function () {
+  e2e.setup()
+  const projectDir: string = path.join('/', 'tmp', '8_read_only_fs')
+
+  beforeEach(() => {
+    if (fs.existsSync(projectDir)) {
+      return
+    }
+
+    // write out the project structure
+    // we cant sue one define by source code since we need to modify the file permissions on the dir and files
+    fs.mkdirSync(path.join(projectDir))
+
+    fs.writeFileSync(path.join(projectDir, 'cypress.json'), `{"projectId": "2pz86o"}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress'))
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'integration'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'integration', 'spec.js'), `it('test',() => expect(true).to.be.true )`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'plugins'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'plugins', 'index.js'), `module.exports = () => {}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'support'))
+    fs.writeFileSync(path.join(projectDir, 'cypress', 'support', 'index.js'), `module.exports = () => {}`)
+    fs.mkdirSync(path.join(projectDir, 'cypress', 'videos'))
+
+    // setup the file/folder permissions
+    // 500 means individual read + execute
+    // the dir needs executable for debugging
+    // the main test here is that it isn't writable, which would be either 6 or 7
+    // See here for more info on these numbers http://permissions-calculator.org/
+    chmodr.sync(projectDir, 0o500)
+
+    // test that we setup the folder structure correctly
+    let err: any
+
+    try {
+      fs.accessSync(projectDir, fs.constants.W_OK)
+    } catch (e) {
+      err = e
+    }
+    expect(err).to.not.be.undefined
+  })
+
+  e2e.it('warns when unable to write to dir', {
+    project: projectDir,
+    spec: 'spec.js',
+    browser: 'chrome',
+    expectedExitCode: 0,
+    snapshot: true,
+  })
```

Comment:
Test projects are automatically scaffolded in `beforeEach`, so you can do all of your `chmodr` stuff via `onRun` on one of the project dirs that already exists:

```js
  e2e.it('warns when unable to write to dir', {
    project: Fixtures.projectPath('e2e'), // copied from `test/support/fixtures/projects/e2e`
    // use a failing spec so we can see a snapshot of the screenshot failing to be written
    // in addition to the video failing to be written
    spec: 'simple_failing_spec.coffee',
    expectedExitCode: 0,
    snapshot: true,
    onRun: (exec) => {
      // ...do the `chmodr` stuff on `Fixtures.projectPath('e2e')` here
      // ...assert that EACCES is thrown here
      // now call `exec` to run the e2e test
      return exec()
      // you may need an `afterEach` call to clean up the permissions
  })
```

This way you don't need to do any of the manual project setup you have in your `beforeEach`.

Reply:
Thanks for the tip, it was driving me crazy that I couldn't do cleanup stuff before the afterEach Fixture code ran. That's why I wasn't using the built in fixture stuff